### PR TITLE
Add Env File resolution

### LINKED ISSUE #169: .env file doesn't get resolved
See https://docs.docker.com/compose/environment-variables/#the-env-file

Example:
image

### PR DESCRIPTION
**Pull request by michaelbayday**
Closes #169 

Add Env file resolution when a .env file is stored in the same folder as the docker compose file.

If there is no .env file stored in the same folder, then the ${variablename}  notation will resume

### CODE REVIEW

**Review by TylerHurtt — APPROVED** [review]
Looks good!

File: src/common/resolveEnvVariables.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+type EnvObject = {
+  [variableName: string]: string;
+};
+
+//this function replaces all env variables within the docker compose yaml string with their values in the env file
+const resolveEnvVariables = (yamlText: string, filePath: string) => {
+  const envFileArray = filePath.split('/');
+  const envFilePath =
+    envFileArray.slice(0, envFileArray.length - 1).join('/') + '/.env';
```
Comment (omertuc) [inline src/common/resolveEnvVariables.ts:11]: Better use https://nodejs.org/api/path.html#path_path_dirname_path and https://nodejs.org/api/path.html#path_path_join_paths for more portable / correct path handling

File: src/common/resolveEnvVariables.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  let envString: string;
+  try {
+    envString = fs.readFileSync(envFilePath).toString();
+  } catch (err) {
+    return yamlText;
+  }
+  let yamlTextCopy = yamlText;
+  console.log(envString);
+  //split by line
+  const envArray = envString.split('\n');
```
Comment (omertuc) [inline src/common/resolveEnvVariables.ts:22]: This is not correct parsing of the .env file. Use https://github.com/motdotla/dotenv for more correct parsing, it's not as trivial as you might think. More specifically, this https://github.com/motdotla/dotenv#can-i-customizewrite-plugins-for-dotenv part might be more helpful

File: src/common/resolveEnvVariables.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  //remove empty last element
+  envArray.splice(-1, 1);
+  //create Object that stores the variable notation to be replace with its value
+  const envObject: EnvObject = envArray.reduce((acc: EnvObject, el: string) => {
+    const [variableName, value] = el.split('=');
+    const variableKey: string = '\\${' + variableName + '}';
```
Comment (omertuc) [inline src/common/resolveEnvVariables.ts:28]: docker-compose.yml also supports the `$VARNAME` syntax, not just the `${VARNAME}` syntax. There's also an option for default values like so: `${VARNAME:5}`

**Review by omertuc — COMMENTED** [review]
There many pitfalls when trying to do all this parsing on your own - maybe it's better to somehow reuse the docker-compose parsing code and try to parse an internal docker-compose representation of this configuration instead?

**Pull request merged**